### DISCUSSION TITLE
Determine what app is currently running?

### DISCUSSION THREAD

**Discussion by traverseda** [SteamDeckHomebrew/decky-loader#159] [Q&A]
So if I want to make a plugin that communicates with a running app, I need some way to determine what app is running and ideally some unique identifier like the apps PID or window ID or something like that.

Is there any easy way to do that in decky-loader?

**Comment by TrainDoctor** [comment]
> So if I want to make a plugin that communicates with a running app, I need some way to determine what app is running and ideally some unique identifier like the apps PID or window ID or something like that.
> 
> Is there any easy way to do that in decky-loader?

We actually have a WIP plugin that does most of the ground work for providing community made performance presets but it's currently not complete. I'll look into exposing it publicly but if you have a prefrered email I can send you the source code if you'd like.

**Reply by traverseda** [reply]
I think I probably won't dig that deeply into it this early, I don't think I realistically have the time to do what I want (integrate things like saving/loading from emulators into the steam menu) any time soon, so I can take my time for that to show up and hopefully mature a bit.

Thanks though!
